src/math/CMakeLists.txt

```diff
@@ -1,58 +1,92 @@
 # SPDX-License-Identifier: BSD-3-Clause
 
 if(BUILD_LIBRARY)
-	add_local_sources(sof numbers.c)
-	return()
+  add_local_sources(sof numbers.c)
+  return()
 endif()
 
-add_local_sources(sof numbers.c)
+set(base_files numbers.c)
 
 if(CONFIG_CORDIC_FIXED)
-        add_local_sources(sof trig.c)
+  list(APPEND base_files trig.c)
 endif()
```

Comment:
can a `add_list_ifdef()` cmake function be used? I'm not a cmake expert, so not sure whether objects are passed by reference

Reply:
@lyakh I was going back and forth with this. Not an cmake expert here either, but given we need new owners for cmake magic in SOF, we probably need to just learn more.

We could add a function, but we'd need to add it as a SOF extension (Zephyr has list_append_if() but we can't use that). 
What I'm personally not sure is that is it worth it to save some lines in definitions versus keeping SOF cmake files as close to standard cmake files as possible. I.e. functions that are used are standard cmake functions. Most of people contributing to SOF will not be cmake experts either.

On the other hand, we have a LOT of use for add_local_sources_if(), so having a replacement macro that facilitates Zephyr/XTOS might still be useful.

Reply:
Update: hmm, it seems add_local_sources_if() is already a "SOF'ism", a macro defined and used only in SOF. So it seems we've already gone the path to use SOF specific macros to avoid repetition. I can update to this approach.